cachix/src/Cachix/Client/Push.hs

```diff
@@ -138,6 +140,18 @@ getCacheAuthToken :: PushSecret -> Token
 getCacheAuthToken (PushToken token) = token
 getCacheAuthToken (PushSigningKey token _) = token
 
+updateProgressBar :: MonadIO m => Maybe ProgressBar -> ConduitM ByteString ByteString m ()
+updateProgressBar Nothing = awaitForever Data.Conduit.yield
+updateProgressBar (Just pg) =
+  await
+    >>= maybe
+      (return ())
+      ( \chunk -> do
+          liftIO $ tickN pg $ BS.length chunk
+          Data.Conduit.yield chunk
```

Comment:
```suggestion
          Data.Conduit.yield chunk
          liftIO $ tickN pg $ BS.length chunk
```

It's only done when it's done. That's assuming the return of `yield` represents completion of that chunk. Probably still not entirely accurate because there's going to be some buffered but unsubmitted input until you flush the compressed stream, which I don't think is something you'd want to do. Anyway, the suggestion seems slightly better.